File: app/src/main/java/com/kidssaveocean/fatechanger/onboarding/OnboardingAdapter.java
package com.kidssaveocean.fatechanger.onboarding;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.kidssaveocean.fatechanger.R;

public class OnboardingAdapter extends FragmentPagerAdapter {
    public static final int NUMBER_OF_ONBOARDING_PAGES = 3;

    public OnboardingAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        int imageId;
        int titleId;
        int subTitleId;
        int descriptionId;
        int layoutId;

        switch (position) {
            case 0:
            default:
                imageId = R.drawable.whale_tail;
                titleId = R.string.welcome_to_fate_changer;
                subTitleId = R.string.children_saving_the_ocean;
                descriptionId = R.string.welcome_to_fate_changer_description;
                layoutId = R.layout.fragment_no_button_onboarding;
                break;
            case 1:
                imageId = R.drawable.actinia;
                titleId = R.string.how_you_can_help;
                subTitleId = R.string.learn_write_share;
                descriptionId = R.string.how_you_can_help_description;
                layoutId = R.layout.fragment_no_button_onboarding;
                break;
            case 2:
                imageId = R.drawable.reef;
                titleId = R.string.stay_engaged;
                subTitleId = R.string.stay_connected_for_updates;
                descriptionId = R.string.stay_engaged_description;
                layoutId = R.layout.fragment_button_onboarding;
                break;
        }

        Bundle bundle = new Bundle();
        Fragment fragment = new OnboardingFragment();
        fragment.setArguments(bundle);
        bundle.putInt(OnboardingFragment.IMAGE, imageId);
        bundle.putInt(OnboardingFragment.TITLE, titleId);
        bundle.putInt(OnboardingFragment.SUBTITLE, subTitleId);
        bundle.putInt(OnboardingFragment.DESCRIPTION, descriptionId);
        bundle.putInt(OnboardingFragment.LAYOUT, layoutId);
        return fragment;
    }

    @Override
    public int getCount() {
        return NUMBER_OF_ONBOARDING_PAGES;
    }
}

File: app/src/main/java/com/kidssaveocean/fatechanger/onboarding/OnboardingAdapter.kt
package com.kidssaveocean.fatechanger.onboarding

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.kidssaveocean.fatechanger.R

//todo fix this
class OnboardingAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    //todo fix this adapter
    override fun getItem(position: Int): Fragment {
        val imageId: Int
        val titleId: Int
        val subTitleId: Int
        val descriptionId: Int
        val layoutId: Int
        var extraId: Int

        when (position) {
            0 -> {
                imageId = R.drawable.imagery_welcome
                titleId = R.string.welcome_to_fate_changer
                subTitleId = R.string.children_saving_the_ocean
                descriptionId = R.string.welcome_to_fate_changer_description
                extraId = R.string.are_you_ready
                layoutId = R.layout.fragment_no_button_onboarding
            }
            1 -> {
                imageId = R.drawable.imagery_help
                titleId = R.string.how_you_can_help
                subTitleId = R.string.learn_write_share
                descriptionId = R.string.how_you_can_help_description
                extraId = R.string.empty
                layoutId = R.layout.fragment_no_button_onboarding
            }
            2 -> {
                imageId = R.drawable.imagery_engage
                titleId = R.string.stay_engaged
                subTitleId = R.string.stay_connected_for_updates
                descriptionId = R.string.stay_engaged_description
                extraId = R.string.empty
                layoutId = R.layout.fragment_button_onboarding
            }
            else -> {
                imageId = R.drawable.imagery_welcome
                titleId = R.string.welcome_to_fate_changer
                subTitleId = R.string.children_saving_the_ocean
                descriptionId = R.string.welcome_to_fate_changer_description
                extraId = R.string.are_you_ready
                layoutId = R.layout.fragment_no_button_onboarding
            }
        }

        val bundle = Bundle()
        val fragment = OnboardingFragment()
        fragment.arguments = bundle
        bundle.putInt(OnboardingFragment.IMAGE, imageId)
        bundle.putInt(OnboardingFragment.TITLE, titleId)
        bundle.putInt(OnboardingFragment.SUBTITLE, subTitleId)
        bundle.putInt(OnboardingFragment.DESCRIPTION, descriptionId)
        bundle.putInt(OnboardingFragment.LAYOUT, layoutId)
        bundle.putInt(OnboardingFragment.EXTRA, extraId)
        return fragment
    }

    override fun getCount() = NUMBER_OF_ONBOARDING_PAGES


    companion object {
        private const val NUMBER_OF_ONBOARDING_PAGES = 3
    }
}
